libs/SalesforceHybridSDK/SalesforceHybridSDK/Classes/SFHybridViewController.m

```diff
@@ -313,7 +317,16 @@ - (void)authenticateWithCompletionBlock:(SFOAuthPluginAuthSuccessBlock)completio
         } else if (failureBlock != NULL) {
             failureBlock(authInfo, error);
         }
-    }];
+    };
+
+    if (![SFUserAccountManager sharedInstance].currentUser) {
```

Comment:
That would be driven through `SalesforceOAuthPlugin` though, which would trigger a different code path.